### DISCUSSION TITLE
Zero-shot modeling unable to predict new documents.

### DISCUSSION THREAD

**Discussion by HaoyuWangEBSCO** [MaartenGr/BERTopic#1810] [Q&A]
I am working on a topic modeling job for customer feedback, and I have found that when models are trained with zero-shot modeling, they become unable to predict or fit.transform new documents. Has anyone else experienced this, or does anyone know a way to avoid this?

**Comment by MaartenGr** [comment]
Have you tried installing BERTopic from its main branch? There are a number of fixes pertaining to zero-shot topic modeling.

**Reply by HaoyuWangEBSCO** [reply]
Hi Maarten, Thank you for your response! By that, do you mean to reinstall the BERTopic and re-training the zero-shot topic modeling model?

**Reply by MaartenGr** [reply]
You can do a `pip install` from the latest commit which should have the fix available. You can find more about that [here](https://stackoverflow.com/questions/13685920/install-specific-git-commit-with-pip). Then, you would indeed re-train the zero-shot topic model.
